packages/clerk-js/src/ui/contexts/ClerkUIComponentsContext.tsx

```diff
@@ -334,39 +341,62 @@ export const useOrganizationSwitcherContext = () => {
   }: {
     organization?: OrganizationResource;
     user?: UserResource;
+  }) => {
+    const redirectUrl = getAfterSelectOrganizationOrPersonalUrl({
+      organization,
+      user,
+    });
+
+    if (redirectUrl) {
+      return navigate(redirectUrl);
+    }
+
+    return Promise.resolve();
+  };
+
+  const getAfterSelectOrganizationOrPersonalUrl = ({
+    organization,
+    user,
+  }: {
+    organization?: OrganizationResource;
+    user?: UserResource;
   }) => {
     if (typeof ctx.afterSelectPersonalUrl === 'function' && user) {
-      return navigate(ctx.afterSelectPersonalUrl(user));
+      return ctx.afterSelectPersonalUrl(user);
```

Comment:
This does make sense to me, but before merging, I'd like to verify that the router the components use internally still updates its state after the setActive invocation. 

To share a bit of context, even though the clerk-js components will use the router of the parent host for navigations (`routerPush` and `routerReplace`) they still hold an internal router state (`clerk-js/src/ui/RouteContext`). This state updates when `navigate` is called but after this change, the internal `navigate` invocation will be replaced by `Clerk#navigate`.

@issuedat apologies, I know the names are confusing here. Tagging @panteliselef as well as he's most likely interested in this as well.